ql/src/java/org/apache/hadoop/hive/ql/parse/repl/load/message/RenameTableHandler.java

```diff
@@ -55,18 +55,19 @@
         }
       }
 
-      String oldName = StatsUtils.getFullyQualifiedTableName(oldDbName, tableObjBefore.getTableName());
-      String newName = StatsUtils.getFullyQualifiedTableName(newDbName, tableObjAfter.getTableName());
+      TableName oldName = TableName.fromString(tableObjBefore.getTableName(), null, oldDbName);
+      TableName newName = TableName.fromString(tableObjAfter.getTableName(), null, newDbName);
       ReplicationSpec replicationSpec = context.eventOnlyReplicationSpec();
       if (ReplUtils.isTableMigratingToTransactional(context.hiveConf, tableObjAfter)) {
         replicationSpec.setMigratingToTxnTable();
       }
-      AlterTableRenameDesc renameTableDesc = new AlterTableRenameDesc(oldName, replicationSpec, false, newName);
+      AlterTableRenameDesc renameTableDesc =
+          new AlterTableRenameDesc(oldName, replicationSpec, false, newName.getNotEmptyDbTable());
```

Comment:
the newName is passed as string; while oldName is a TableName (ok in a followup as well)